Source/Core/DiscIO/Volume.h

```diff
@@ -51,6 +51,12 @@ class Volume
       return {};
     return Common::FromBigEndian(temp);
   }
+  std::optional<u64> ReadSwappedAndShifted(u64 offset, const Partition& partition) const
+  {
+    const std::optional<u32> temp = ReadSwapped<u32>(offset, partition);
+    return temp ? static_cast<u64>(*temp) << GetOffsetShift() : std::optional<u64>();
```

Comment:
It doesn't seem to work when used inside a ternary operator like this. I get the following error (in VS2017):

```
error C2446: ':': no conversion from 'const std::nullopt_t' to 'u64'
```